libraries/AC_Avoidance/AP_OADijkstra.cpp

```diff
@@ -245,6 +326,210 @@ bool AP_OADijkstra::check_inclusion_polygon_updated() const
     return (_inclusion_polygon_update_ms != fence->polyfence().get_inclusion_polygon_update_ms());
 }
 
+// intersect two sets of polygons as defined by points, index and number. Record intersectons in provided array
+// self bool should be set if serching for intersections between same group of polygons
+bool AP_OADijkstra::intersect_polygon_groups(AP_ExpandingArray<Vector2f> &points1, AP_ExpandingArray<polygon_start_finish> &index1, uint8_t num_poly1, AP_ExpandingArray<Vector2f> &points2,  AP_ExpandingArray<polygon_start_finish> &index2, uint8_t num_poly2, AP_ExpandingArray<Vector2f> &intersections, uint8_t &num_intersections, bool self)
+{
+    if (!intersections.expand_to_hold(1)) {
+        return false;
+    }
+    num_intersections = 0;
+
+    if (num_poly1 == 0 || num_poly2 == 0) {
+        // no polygons to intersect
+        return true;
+    }
+
+    // loop through polygon groups
+    for(uint8_t i = 0; i<num_poly1 - (self?1:0); i++) {
```

Comment:
we should have a space between "for" and "("

Reply:
fixed